puppet/zulip/files/nginx/zulip-include-frontend/app

```diff
@@ -31,8 +31,6 @@ location /static/webpack-bundles/ {
 location /json/events {
     proxy_pass $tornado_server;
     include /etc/nginx/zulip-include/proxy_longpolling;
-
-    proxy_set_header X-Real-IP       $remote_addr;
 }
```

Comment:
This nicely simplifies this block to not have extra settings outside the standard `proxy_longpolling` module, which is great for readability.

Thinking about this question for the rest of the file, should the `uwsgi_param` lines be in
`puppet/zulip/files/nginx/uwsgi_params` instead?

I think that change would affect `puppet/zulip/files/nginx/zulip-include-maybe/uploads-route.*`; I'm not sure whether that would be might be a good thing, in at least 1/2 cases?

```
  $no_serve_uploads = zulipconf('application_server', 'no_serve_uploads', '')
  if $no_serve_uploads != '' {
    # If we're not serving uploads locally, set the appropriate API headers for it.
    $uploads_route = 'puppet:///modules/zulip/nginx/zulip-include-maybe/uploads-route.noserve'
  } else {
    $uploads_route = 'puppet:///modules/zulip/nginx/zulip-include-maybe/uploads-route.internal'
  }
```

Reply:
Maybe something like:

```
# Configuration for how uploaded files are served.  The default is to server uploaded files using using the `nginx`
# `internal` feature via django-sendfile2, which basically does an internal redirect and returns the file content
# from nginx in an HttpResponse that would otherwise have been a redirect.
#
# For installations using S3 to serve uploaded files, we want Django to handle the /serve_uploads route, so
# that it can do authentication and serve a redirect.
```

But looking at this, what I'm puzzled by is why the `location /user_uploads` part lives in `/etc/nginx/zulip-include/uploads.route` rather than the main `puppet/zulip/files/nginx/zulip-include-frontend/app`.  

5614d51afcaf09dca6f22dfc36303b8275610608 is the history.

Reply:
Looks like the split was necessary until 33c941407b72a320a1975947bdc002243af794a0.  I'll clean it up.

Reply:
Yeah, I had a vague recollection that there were originally 3 versions and that's why it was a little crazy.

Thanks for cleaning this up!  It really does feel more pleasant to read in the revised version.